sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -129,6 +131,46 @@ private[connect] object PipelinesHandler extends Logging {
     }
   }
 
+  /**
+   * Block unsupported SQL commands that are not explicitly allowlisted.
```

Comment:
Can we explain the spirit of what's allowed here? I.e. basically we want to limit all plans that have side effects, because registering pipeline definitions should be side effect free. While in general we can't necessarily stop all side effects from slipping through the cracks, it's a best effort to avoid users shooting themselves in the foot.

Reply:
Done!